c++/src/kj/common.h

```diff
@@ -2278,6 +2278,23 @@ struct ByteLiteral<1ul> {
 
 }
 
+class ThreadId {
+// Unique thread identifier.
+// Implemented as thread_local address, so could be efficient even for production
+// environments especially with LTO.
+
+public:  
+  ThreadId();
+  // Initializes to current thread id
```

Comment:
IMO default constructor should probably initialize to null, or should be deleted.

We should have a static `ThreadId::current()` to get the current thread.